cubano-webdriver/src/main/java/org/concordion/cubano/driver/web/provider/RemoteBrowserProvider.java

```diff
@@ -120,6 +116,83 @@ public WebDriver createDriver() {
         }
     }
 
+    private boolean isNullOrEmpty(String string) {
+        return string == null || string.isEmpty();
+    }
+    
+    private URL getProxyUrl() {
+    	WebDriverConfig config = WebDriverConfig.getInstance();
+    	
+        String proxyInput = isNullOrEmpty(config.getProxyHost()) ? System.getenv("HTTPS_PROXY") : config.getProxyHost();
+        if (isNullOrEmpty(proxyInput)) {
+            return null;
+        }
+        try {
+            return new URL(proxyInput.matches("^http[s]?://.*$") ? proxyInput : "http://" + proxyInput);
+        } catch (MalformedURLException e) {
+        	// TODO
+//            log.error("Invalid proxy url {}", proxyInput, e);
+            return null;
+        }
+    }
+
+    private final BasicCredentialsProvider createBasicCredentialsProvider(URL proxyURL) {
+    	if (proxyURL == null) {
+            return null;
```

Comment:
Is this Java 8 and above or still supporting older versions? If 8 and above, consider using [Optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) rather than returning null.

Reply:
Needs to return null as the return value is passed to:
builder.setDefaultCredentialsProvider(createBasicCredentialsProvider(proxyURL));

I've added some todo's here so we can check envrionment variable usage for proxy settings, although I'm tempted to ignore the issue and stick with same settings that webdrivermanager supports.